pkg/pulumiyaml/analyser.go

```diff
@@ -1085,9 +1087,13 @@ func (tc *typeCache) typeConfig(r *Runner, node configNode) bool {
 	}
 	// check for incompatible types between config/ configuration
 	if typExisting, ok := tc.configuration[k]; ok {
-		if typMatch, ok := unifyConfigType(typExisting, typCurrent); !ok {
+		var curVal interface{}
+		if isConfigEnv {
+			curVal = v
+		}
+		if typMatch, ok := unifyConfigType(typExisting, typCurrent, curVal); !ok {
```

Comment:
Is there a reason not to pass `node.value()` in all cases here?
```suggestion
		if typMatch, ok := unifyConfigType(typExisting, typCurrent, node.value()); !ok {
```

Reply:
`node.value()` for a `configuration` key would be a `*ast.ConfigParamDecl`, but I realize now that's irrelevant. Also I believe we only hit conflicts where we are registering a `config` key but already have a matching `configuration` key since we store in-memory `configuration` then `config` keys`.